taipy/core/cycle/_cycle_manager.py

```diff
@@ -150,3 +150,14 @@ def _get_cycles_by_frequency_and_overlapping_date(
     @classmethod
     def _get_cycles_cdt(cls, cdt: Callable[[Cycle], bool], cycles: List[Cycle]) -> List[Cycle]:
         return [cycle for cycle in cycles if cdt(cycle)]
+
+    @classmethod
+    def _get_scenarios_by_time(cls , at: datetime):
+        from ..scenario._scenario_manager_factory import _ScenarioManagerFactory
+
+        scenarios = []
+        for scenario in _ScenarioManagerFactory._build_manager()._get_all():
+            cycle = scenario.cycle
+            if cycle.start_date <= at and at <=  cycle.end_date:
+                scenarios.append(scenario)
+        return scenarios
```

Comment:
The method is in CycleManager now, which is great.

However, you are not utilizing any method of this class but still calling to the `_ScenarioManager._get_all()`, which results in a local import.

In the _CycleManager, you can see that we have `_get_cycles_by_frequency_and_overlapping_date()` and `_get_cycles_by_frequency_and_start_date()` methods. Probably you can use this 2 methods, or create something similar.

Then, on `taipy/core/taipy.py`, you can expose a `get_scenarios_by_time` method, which get cycles by the `at` parameter, then get scenario.